### PR TITLE
Script for building openssl and supla-client for iOS improved

### PR DESCRIPTION
**Pull request by mipolansk**

### CODE REVIEW

File: supla-client/Tools/build_ios.sh [hunk 1]
```diff
@@ -16,78 +16,124 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
 
-SDK_VERSION="9.0"
+SDK="9.0"
```
Comment (przemyslawzygmunt) [inline supla-client/Tools/build_ios.sh:19]: Change to 16.2

File: supla-client/Tools/build_ios.sh [hunk 2]
```diff
@@ -16,78 +16,124 @@
...
+  RC=$?
+  if [ $RC -ne 0 ]
+  then
+    echo "[ERROR] Make failed with rc=${RC}, please check logs ${LOG_FILE}"
+    exit 1
+  fi
+
+  mv libsupla-client.a "${PLATFORM}_${ARCH}/"
+}
+
```
Comment (przemyslawzygmunt) [inline supla-client/Tools/build_ios.sh:101]: Add:

mkdir -p ${BUILD_DIR}/include
cp ../src/supla-client.h ${BUILD_DIR}/include/
cp ../../supla-common/proto.h ${BUILD_DIR}/include/

File: supla-client/Tools/build_ios.sh [hunk 3]
```diff
@@ -16,78 +16,124 @@
...
-echo "${BUILD_DIR}/libsupla-client.a"
+  "${BUILD_DIR}/iPhoneSimulator_arm64/libsupla-client.a" \
+  "${BUILD_DIR}/iPhoneSimulator_x86_64/libsupla-client.a" \
+  -create -output "${BUILD_DIR}/libsupla-client-iphonesimulator.a"
+
+echo "[INFO] Creating XCFramework"
+rm -rf "${BUILD_DIR}/LibSuplaClient.xcframework"
+xcodebuild -create-xcframework \
+  -library "${BUILD_DIR}/libsupla-client-iphoneos.a" \
+  -library "${BUILD_DIR}/libsupla-client-iphonesimulator.a" \
```
Comment (przemyslawzygmunt) [inline supla-client/Tools/build_ios.sh:124]: Add:
  -headers "${BUILD_DIR}/include" \

File: supla-client/Tools/build_ios.sh [hunk 4]
```diff
@@ -16,78 +16,124 @@
...
+  echo "[INFO] XCFramework created: ${BUILD_DIR}/LibSuplaClient.xcframework"
+fi
+
+rm -rf "${BUILD_DIR}/iPhoneOS_armv7" \
+  "${BUILD_DIR}/iPhoneOS_armv7s" \
+  "${BUILD_DIR}/iPhoneOS_arm64" \
+  "${BUILD_DIR}/iPhoneSimulator_arm64" \
+  "${BUILD_DIR}/iPhoneSimulator_x86_64" \
+  "${BUILD_DIR}/libsupla-client-iphoneos.a" \
+  "${BUILD_DIR}/libsupla-client-iphonesimulator.a"
```
Comment (przemyslawzygmunt) [inline supla-client/Tools/build_ios.sh:139]: Add:

\
"${BUILD_DIR}/include"

File: supla-client/Tools/openssl/build_openssl.sh [hunk 5]
```diff
@@ -1,132 +1,165 @@
...
+# SDK8.2 (XCode 6.2)
 
-# Credits:
-# https://github.com/st3fan/ios-openssl
-# https://github.com/x2on/OpenSSL-for-iPhone/blob/master/build-libssl.sh
+set -e
 
-#SDK8.2 (XCode 6.2)
+# Defaults
+SDK="8.2"
```
Comment (przemyslawzygmunt) [inline supla-client/Tools/openssl/build_openssl.sh:9]: Change to 16.2

File: supla-client/Tools/build_ios.sh (hunk 1 above)
Comment (mipolansk) [reply supla-client/Tools/build_ios.sh]: Done

File: supla-client/Tools/build_ios.sh (hunk 2 above)
Comment (mipolansk) [reply supla-client/Tools/build_ios.sh:101]: I added it once after building all versions

File: supla-client/Tools/build_ios.sh (hunk 3 above)
Comment (mipolansk) [reply supla-client/Tools/build_ios.sh:128]: done

File: supla-client/Tools/build_ios.sh (hunk 4 above)
Comment (mipolansk) [reply supla-client/Tools/build_ios.sh]: done

File: supla-client/Tools/openssl/build_openssl.sh (hunk 5 above)
Comment (mipolansk) [reply supla-client/Tools/openssl/build_openssl.sh]: done

**Review by przemyslawzygmunt — APPROVED** [review]

**Pull request merged**